tests/components/hyperion/test_light.py

```diff
@@ -27,98 +27,84 @@ def create_mock_client():
+    await setup_entity(hass, client=client)
 
     # Make sure the background task is initiated.
     assert client.start_background_task.called
-
-    # Make sure the entity is added.
-    assert add_entities.called
+    assert hass.states.get(TEST_ENTITY_ID) is not None
 
 
 async def test_setup_platform_not_ready(hass):
     """Test the platform not being ready."""
-    add_entities = Mock()
     client = create_mock_client()
     client.async_connect = CoroutineMock(return_value=False)
 
-    with pytest.raises(PlatformNotReady):
-        with patch("hyperion.client.HyperionClient", return_value=client):
-            await hyperion_light.async_setup_platform(hass, TEST_CONFIG, add_entities)
-            await hass.async_block_till_done()
+    await setup_entity(hass, client=client)
 
     # Make sure the background task is initiated.
     assert not client.start_background_task.called
-
-    # Make sure the entity is added.
-    assert not add_entities.called
+    assert hass.states.get(TEST_ENTITY_ID) is None
 
 
 async def test_light_basic_properies(hass):
     """Test the basic properties."""
-    client, entity = create_client_and_entity()
-    assert entity
-    assert not entity.should_poll
-    assert entity.name == TEST_NAME
-    assert entity.brightness == 255
-    assert entity.hs_color == (0.0, 0.0)
-
-    client.is_on.return_value = False
-    assert not entity.is_on
-
-    client.is_on.return_value = True
-    assert entity.is_on
+    client = create_mock_client()
+    await setup_entity(hass, client=client)
 
-    assert entity.icon == hyperion_light.ICON_LIGHTBULB
-    assert entity.effect == hyperion_light.KEY_EFFECT_SOLID
+    entity = hass.states.get(TEST_ENTITY_ID)
+    assert entity.state == "on"
+    assert entity.attributes["brightness"] == 255
+    assert entity.attributes["hs_color"] == (0.0, 0.0)
+    assert entity.attributes["icon"] == hyperion_light.ICON_LIGHTBULB
+    assert entity.attributes["effect"] == hyperion_light.KEY_EFFECT_SOLID
 
     # By default the effect list is the 3 external sources + 'Solid'.
-    assert len(entity.effect_list) == 4
-
-    assert entity.supported_features == hyperion_light.SUPPORT_HYPERION
-
-    client.is_connected = False
-    assert not entity.available
-
-    client.is_connected = True
-    assert entity.available
+    assert len(entity.attributes["effect_list"]) == 4
 
-    unique_id = "1234"
-    client.id = unique_id
-    assert entity.unique_id == unique_id
+    assert entity.attributes["supported_features"] == hyperion_light.SUPPORT_HYPERION
 
 
 async def test_light_async_turn_on(hass):
     """Test turning the light on."""
-    client, entity = create_client_and_entity()
+    client = create_mock_client()
+    await setup_entity(hass, client=client)
 
     # On (=), 100% (=), solid (=), [255,255,255] (=)
     client.async_send_set_color = CoroutineMock(return_value=True)
-    await entity.async_turn_on()
+    await common.async_turn_on(hass, TEST_ENTITY_ID)
```

Comment:
Please call `hass.services.async_call` directly. The service helpers in common are legacy.